packages/graphql/src/entities/GitInfo.ts

```diff
@@ -0,0 +1,35 @@
+import { nxs, NxsResult } from 'nexus-decorators'
+import type { BaseContext } from '../context'
+import type { GitInfo as GetGitInfo } from '@packages/server/lib/util/git'
+
+@nxs.objectType({
+  description: 'Git info about a file',
+})
+export class GitInfo implements GetGitInfo {
+  private config: GetGitInfo | null
+
+  constructor (file: string, ctx: BaseContext) {
+    this.config = ctx.actions.getGitInfo(file)
```

Comment:
I've been seeing this and I think we need to sync up to clarify the distinction of Actions. Getting something shouldn't be considered an action - actions should only be called from mutations and should have a side-effect.

Going to add a guard soon so `ctx.actions` cannot be accessed from within a query.

Reply:
Hm ok - I am using the actions abstraction so can read the git data from the file system - isn't this considered a side effect (reading stdout from a process executing git, for example)? How would you suggest structuring this (eg - an action that needs to execute a process and return the result from stdout)

Reply:
I think it depends on how you define side-effect... in theory, everything is a side effect. In practice/practical terms I consider a side-effect something that mutates information. So I would not consider `git log` a side-effect since it's read-only.

Also I'd probably consider querying for & caching this data independent of the GraphQL entity layer, this sort of git watching thing would probably kick off as a background process & update the data centrally (provided into the Context) and the resolvers would read from that

Reply:
Let's sync early next week. There are some cases I am not clear on how best to handle, based on this definition of a side effect. Eg: when you query for a project [see this PR](https://github.com/cypress-io/cypress/pull/18113) we also update the cache on the filesystem. 

So far I've been using actions as a "this is where the logic goes; server actions can touch the filesystem, etc, and their test implementation can be mocked". I may have misunderstood your original intention w/ the actions abstraction.

Reply:
Include me in this sync as well, it would be good if the three of us were aligned on the architecture of the graphql server. On top of discussing actions I think we could talk about how we want to structure our nodes as well.